language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/visitors/FunctionCallExpressionTypeFinder.java

```diff
@@ -181,6 +181,9 @@ public void visit(StartActionNode startActionNode) {
             FutureTypeSymbol futureTypeSymbol = (FutureTypeSymbol) returnTypeSymbol;
             TypeSymbol typeSymbol = futureTypeSymbol.typeParameter().orElse(null);
             checkAndSetTypeResult(typeSymbol);
+        } else {
+            TypeSymbol nilTypeSymbol = semanticModel.types().NIL;
```

Comment:
Earlier we were getting `null` for the returnTypeSymbol? Asking for my understanding for this change.

Reply:
Yes and if we apply the `Create function` code action to the following scenario where the function doesn't have a type, 
```
public function main (string... args) {
   () value = getValue();
}
```
The getValue function was created with `returns ()`. These changes were added to avoid that.